staging/src/k8s.io/client-go/tools/clientcmd/client_config.go

```diff
@@ -208,7 +208,9 @@ func getServerIdentificationPartialConfig(configAuthInfo clientcmdapi.AuthInfo,
 	// configClusterInfo holds the information identify the server provided by .kubeconfig
 	configClientConfig := &restclient.Config{}
 	configClientConfig.CAFile = configClusterInfo.CertificateAuthority
-	configClientConfig.CAData = configClusterInfo.CertificateAuthorityData
+	if len(configClientConfig.CAFile) == 0 {
```

Comment:
I don't think this is the correct place to do conditional copying... also, this breaks the documented precedence of the CertificateAuthorityData field, which is that it overrides CertificateAuthority